src/fathom.js

```diff
@@ -0,0 +1,189 @@
+ * @typedef {Object} FathomDataObject
+ *
+ * TODO: Finalize data schema
+ * Likely would involve the URL, pageID, and the fathom classification results.
+ * Currently, just sends an object containing every element with greater than
+ * 0.5 confidence score (key), along with their feature vector and actual score (value).
+ */
+
+/**
+ * The listener specified by the client.
+ * @callback fathomDataListener
+ * @param {FathomDataObject} Results of the Fathom classification on a page.
+ */
+
+/**
+ * @typedef {Object} FathomDataListenerRecord
+ * @property {matching.MatchPatternSet} matchPatternSet - The match patterns for the listener.
+ * @property {browser.contentScripts.RegisteredContentScript} contentScript - The content
+ * script associated with the listener.
+ */
+
+/**
+ * A map where the key is a listener function and each value is a record for that listener function.
+ * @constant {Map<fathomDataListener, FathomDataListenerRecord>}
+ * @private
+ */
+const fathomDataListeners = new Map();
+
+/**
+ * An event that fires when a page's contents has been successfully classified
+ * through a Fathom content script. 
+ *
+ * @constant onFathomData
+ */
+export const onFathomData = events.createEvent({
+    name: "webScience.fathom.onFathomData",
+    addListenerCallback: addListener,
+    removeListenerCallback: removeListener,
+    notifyListenersCallback: () => { return false; }
+});
+
+/**
+ * Whether the module has completed initialization.
+ * @type{boolean}
+ * @private
+ */
+let initialized = false;
+
+
+/**
+ * A callback function to add a fathomDataListener.
+ * @param {fathomDataCallback} listener - The listener that is being removed.
+ * @param {Object} options - Options for the listener.
+ * @param {string[] options.matchPatterns} matchPatterns - The match patterns 
+ * for pages where the listener should be notified.
+ * @private
+ */
+async function addListener(listener, {matchPatterns}) {
+    // Initialize the listener
+    if (!initialized) {
+        initialized = true;
+
+        // Initialize pageManager
+        await pageManager.initialize();
+        
+        // Listen for Fathom classification results, sent by content script
+        // This messageListener receives a fathomDataObject from a content script.
+        messaging.onMessage.addListener(messageListener,
+            {
+                type: "webScience.fathom.fathomData",
+                schema: {
+                    results: "object"
+                }
+            }
+        );
+        
+        // Message to send to content script if this page should be classified
+        messaging.registerSchema("webScience.fathom.isClassifiable", {
+            isClassifiable: "boolean",
+        })
+
+        // When a tab is updated, send it a message if the page should be 
+        // classified with Fathom
+        // TODO: onUpdated may not be the right event
+        browser.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
```

Comment:
Would use the `pageManager.onPageVisitStart` event so you can include a `pageId` in the message.